src/Parser/TypeSchema/BCLayer.php

```diff
@@ -52,6 +52,10 @@ public static function transformDefinition(\stdClass $data): \stdClass
             $data->type = 'struct';
         }
 
+        if (isset($data->{'x-psx-mapping'})) {
+            $data->{'x-psx-mapping'} = (array) $data->{'x-psx-mapping'};
+        }
+
```

Comment:
Hey @AMar4enko thanks for the PR, in general `x-psx-mapping` is only a `psx/schema` specific implementation detail which is not available at the [TypeSchema](https://typeschema.org/) spec. But we can of course also parse this, could you explain a little bit why this is needed, the specific attributes are parsed at s. `psx-schema/src/Parser/TypeSchema.php:164` so it should already work?